esp-alloc/src/lib.rs

```diff
@@ -61,35 +67,100 @@ use core::{
 };
 
 use critical_section::Mutex;
+use enumset::{EnumSet, EnumSetType};
 use linked_list_allocator::Heap;
 
+/// The global allocator instance
+#[global_allocator]
```

Comment:
Could this attribute be behind a feature? Seems like esp-wifi will be using the instance directly rather than via the global allocator.
I use a custom heap implementation and also use WiFi 😅, so this attribute would prevent me from upgrading.

Reply:
`esp-wifi` won't use it directly - it's just to make it easy to implement `free_internal_heap` and `allocate_from_internal_ram` inside `esp-alloc`

In your case if you use your own allocator and you implement the two functions it should work fine for you. 

I guess if someone uses `esp-alloc` they never need to opt out of the global allocator?

Reply:
> I guess if someone uses `esp-alloc` they never need to opt out of the global allocator?

I'm still making use of `EspHeap` in my custom implementation.

> won't use it directly

Poor choice of words on my part. I meant it won't use the `GlobalAlloc` trait, which means there's no need `#[global_allocator]`.

I suppose I can just copy the `EspHeap` implementation and not depend on `esp-alloc`. Then I can implement those two functions to support `esp-wifi`.
I'd only request adding a comment that user's are free to implement the function to prevent anyone from hard coding it in future.